components/OAuth2.0/authorization.c

```diff
@@ -0,0 +1,423 @@
+#include "OauthHttpLocalServer.h"
+#include "SpiffsManger.h"
+#include "OauthAPICall.h"
+#include "cJSON.h"
+
+// ****************************** Global Variables
+OAuthInterfaceHandler_t *AuthInterfaceHandler;
+OAuthPrivateHandler_t AuthPrivateHandler;
+
+// ****************************** Local Variables
+static const char *TAG = "OAuthTask";
+
+/**
+ * @brief This function searches for specific patterns ('code' and 'state') within a character 
+ *        array and returns a boolean value indicating if either pattern was Found.
+ * @param[in] Response The character array to search within, 
+ *            and Response is response from first stage from Service athurisiation
+ * @param[in] SizeRes The size of the character array.
+ * @return Returns true if either the 'code' or 'state' pattern was Found, and false otherwise.
+ */
+static bool FindCode(char *Response, uint16_t SizeRes)
+{
+    char *codeString = {"code"};
+    uint16_t codeLength = strlen(codeString);
+
+    if (Response == NULL || SizeRes < codeLength)
+    {
+        // Invalid input, either null pointer or insufficient buffer size
+        return false;
+    }
+
+    for (uint16_t i = 0; i <= SizeRes - codeLength; ++i)
+    {
+        bool Found = true;
+        for (uint16_t j = 0; j < codeLength; ++j)
+        {
+            if (Response[i + j] != codeString[j])
+            {
+                Found = false;
+                break;
+            }
+        }
+        if (Found)
+        {
+            return true; // Found the access token substring
+        }
+    }
+    return false; // Access token substring not Found
+}
+
+/**
+ * @brief This function handles the first HTTPS request to Service and 
+ *        redirects the user to the authorization page.
+ * @param[in] req The HTTP request object.
+ * @return Returns ESP_OK if the request is processed successfully.
+ */
+static esp_err_t RequestDataAccess(httpd_req_t *HttpdRequest)
+{
+    char *localURL;
+    localURL = (char *)malloc((2 * SMALL_BUF) * sizeof(char));
+    if (localURL == NULL)
+    {
+        ESP_LOGI(TAG, "Failed to allocate memory for the array.\n\n");
+    }
+    memset(localURL, 0x0, SMALL_BUF * 2);
+    ESP_LOGI(TAG, "Starting authorization, sending request for TOKEN");
+    sprintf(localURL, AuthInterfaceHandler->ClientConfig.requestURL, 
+            AuthInterfaceHandler->ClientConfig.clientID, AuthInterfaceHandler->ClientConfig.redirectURL);
+    httpd_resp_set_hdr(HttpdRequest, "Location", localURL);
+    httpd_resp_set_type(HttpdRequest, "text/plain");
+    httpd_resp_set_status(HttpdRequest, "302");
+    httpd_resp_send(HttpdRequest, "", HTTPD_RESP_USE_STRLEN);
+    free(localURL);
+    return ESP_OK;
+}
+
+/**
+ * @brief This function handles the callback HTTPS request from Service and processes the response data.
+ * @param[in] req The HTTP request object.
+ * @return Returns ESP_OK if the request is processed successfully.
+ */
+static esp_err_t HttpsCallbackHandler(httpd_req_t *HttpdRequest)
```

Comment:
please replace it in to the OauthHttpLocalServer.c

Reply:
done.